### DISCUSSION TITLE
How to refresh a Token in RU

### DISCUSSION THREAD

**Discussion by twwolff** [Digitaler-Impfnachweis/certification-apis#124] [Q&A] [answered]
I try to refresh the token but it is not working (server always close the connection without answer). The endpoint I am calling : "https://id.ru.impfnachweis.info/auth/realms/ubirch-certify/protocol/openid-connect/token"
The documentation is not helpfull. 
I succesfully use the same endpoint to get the initial token.

I set this parameter in the url Form encoded body :
Parameter | Value
-- | --
client_id | user-access-ti
redirect_uri | connector://authenticated
client_secret | some_secret12345
grant_type | refresh_token
refresh_token | "refresh-token-from-previous-request"
scope | openid

Is that ok ? Am i making some mistakes ?
Thank for any help.

**Comment by nicolaslang1** [comment]
Hi, i am testing on the Public API 

and i use the following config:

<html>
<body>


Parameter | Value
-- | --
client_id | user-access-ti
client_secret | some_secret12345
grant_type | refresh_token
refresh_token | "refresh-token-from-previous-request"
scope | openid profile


</body>
</html>
And the token endpoint returns my token with no changes. So the expiration date doesn't increase and the code doesn't change so it has no effect. So in the end the api answers my request but it doesn't refresh my token...

**Reply by nicolaslang1** [reply]
@thinkberg

**Reply by thinkberg** [reply]
This is a topic, that @oliver-steinbrecher is better suited to answer.

**Comment by oliver-steinbrecher** [comment, marked as answer]
Please see https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokens

**Reply by twwolff** [reply]
Thank but i have checked this link before i ask the question. But the link is a global specification for the open-id token refresh and this do not specify which client_id, scope and redirect_ui we must use. Thank in Advance.
